src/Type/Php/ArrayFilterFunctionReturnTypeReturnTypeExtension.php

```diff
@@ -52,4 +65,42 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
 		return new ArrayType($keyType, $itemType, true);
 	}
 
+	private function removeFalsey(Type $type): Type
+	{
+		$falseyTypes = new UnionType([
+			new NullType(),
+			new ConstantBooleanType(false),
+			new ConstantIntegerType(0),
+			new ConstantFloatType(0.0),
+			new ConstantStringType(''),
+			new ConstantArrayType([], []),
+		]);
+
+		if ($type instanceof ConstantArrayType) {
```

Comment:
Please also work with union of constant arrays - `array()|array(...)|array(...)`... They can be created by having a literal array and then adding a key for example in an `if` branch. This condition should state `count(TypeUtils::getConstantArrays($type)) > 0` instead. I'm not sure how to implement the logic though...

Reply:
Fixed